test/model/test_store.dart

```diff
@@ -158,6 +163,9 @@ class TestGlobalStore extends GlobalStore {
     );
     updateMachines[accountId] = UpdateMachine.fromInitialSnapshot(
       store: store, initialSnapshot: initialSnapshot);
+    if (loadPerAccountPostDuration != null) {
+      await Future<void>.delayed(loadPerAccountPostDuration!);
+    }
     return Future.value(store);
```

Comment:
Hypothetically, this can happen when `removeAccount` was called
    during `doLoadPerAccount`, possibly due to the user logging out
    from choose-account page while the store is getting reloaded.
    However, it is currently not reachable because of a bug in
    `UpdateMachine.load`.  See #1354.

We already have a test that reaches the analogous situation when initially loading an account: https://github.com/zulip/zulip-flutter/issues/1354#issuecomment-2658121787

So we should be able to use the same technique to test this change, without adding a new complication to TestGlobalStore. You'll see in that that existing test case that it uses the subclass LoadingTestGlobalStore, which gives the test much more direct hands-on control of how `doLoadPerAccount` behaves.

Reply:
Thanks! That's really helpful. The navigator tests are kept as-is, because they use `ZulipApp`, but the more bare-bone tests in `test/model/store_test.dart` are all switched to use this technique.